### DISCUSSION TITLE
Add kidex serch results to overview and anyrun.

### DISCUSSION THREAD

**Discussion by Nevrdid** [end-4/dots-hyprland#297] [Ideas]
Hi, i would like to know if you are interested to add kidex results to launchers and also get some help to finish it on my side.
I already succesfully include it into anyrun and i'm actually trying to add it to ags's overview without success.
Currently i'm trying something like this in `ags/modules/overview/windowcontent.js`: SearchAndWindows->onChange:
```
if (text[0] == 's' && text[1] == ' ') { // Search
                var command = `bash -c "kidex-client get-index | grep ${text.substring(2)} | awk 'FS=\"\\\"\" {print $4}'"`;
                print(command);
                var contents = exec(command);
                ...
}
```
The command work well when i directly launch it but the content is empty there.
(Js isn't a language i'm used to so please don't blame me to get stuck here x)
Thanks for the possibly help and for that great config you offer.

Update:
Here is a working result but it deserve some more work i guess. Like writing the first characters of each parents folder as a prompt would does it and a distinction between files and folder. 
Anyway, that's a good start for a quick regexp search even if i didn't succed to config kidex for fully index user's home content.
```
            if (text[0] == 's' && text[1] == ' ') { // Search
                var command = `bash -c 'kidex-client get-index | grep -E ${text.substring(2)} | cut -d "\\\"" -f 4'`;
                const contents = exec(command);
                const files = contents.split("\n").slice(1, 10);
                files.forEach((item) => {
                    var file = Gio.File.new_for_path(item);
                    resultsBox.add(DirectoryButton({parentPath: file.get_parent().get_path(), name: file.get_basename()}));
                })
            }
```

**Comment by end-4** [comment]
I'm not interested in having a background file indexer
About "a distinction between files and folder.", maybe sort the results? At least that would make a folder appear on top of its children. Correct me if I'm wrong

**Reply by Nevrdid** [reply]
Well i guess the concern is about the usage of the service as i know baloo and tracker are really not light solution and i usually disable them. Tho, i tried kidex and i no more want to disable it as it looks really thrifty. 
Anyway, i don't plan to fork the project until i find a must-have to add so if anyone want me to share my solution, just ask.
I also added a firefox history and bookmarks lookup written by hand as the others scripts i found were not working.
Have a nice day.
